client/application_service.go

```diff
@@ -91,6 +93,47 @@ func (service *ApplicationService) sendTask(app *Application, jobType string, ta
 		return err
 	}
 
-	_, err = service.Do(req)
+	var taskResp TaskResponse
+	_, err = service.DoWithResponse(req, taskResp)
+	if err != nil {
+		return err
+	}
+
+	req, err = service.NewRequest("GET", taskResp.Ref)
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(3 * time.Second)
+	defer ticker.Stop()
+	done := make(chan bool)
+
+	go func() {
+		var execution TaskExecution
+		_, err = service.DoWithResponse(req, execution)
+		if err != nil {
+			log.Printf("[ERROR] Error on execute request to check task status. %s", err)
+			done <- true
+		}
+
+		if execution.Status == "ERROR" {
+			err = errors.New("Error on execute tasks")
```

Comment:
same here, are we only calling this once or do we want to do something like an exponential backoff?